app/utils/rentry/wrapper.py

```diff
@@ -1,85 +1,88 @@
+        response.data = response.text
+        return response
 
-class RentryUpload:
-    """Uploader class to attempt to upload data to Rentry.co"""
+    def get_csrf_token(self):
+        # Get CSRF token from the response cookies after making a GET request to the base URL
+        response = self.get(BASE_URL)
+        return response.cookies.get("csrftoken")
 
+class RentryUpload:
     def __init__(self, text: str):
-        response = self.new(text)
-        if response["status"] != "200":
-            self.upload_success = False
-            self.url = None
-            logger.error("error: {}".format(response["content"]))
-            try:
-                for i in response["errors"].split("."):
-                    i and logger.warning(i)
-                logger.error("RentryUpload failed!")
-            except:
-                logger.error("RentryUpload failed!")
-        else:
-            self.upload_success = True
-            self.url = response["url"]
-            logger.debug("RentryUpload successfully uploaded data!")
-            logger.debug("Url: {}".format(self.url))
-            logger.debug("Edit code: {}".format(response["edit_code"]))
+        self.upload_success = False
+        self.url = None
+
+        try:
+            response = self.new(text)
+            if response.get("status") != "200":
+                self.handle_upload_failure(response)
+            else:
+                self.upload_success = True
+                self.url = response["url"]
+        finally:
+            if self.upload_success:
+                logger.debug(
+                    f"RentryUpload successfully uploaded data! Url: {self.url}, Edit code: {response['edit_code']}")
+
+    def handle_upload_failure(self, response):
+        error_content = response.get("content", "Unknown")
+        errors = response.get("errors", "").split(".")
+        logger.error(f"Error: {error_content}")
+        for error in errors:
+            error and logger.warning(error)
+        logger.error("RentryUpload failed!")
 
     def new(self, text):
-        client, cookie = UrllibClient(), SimpleCookie()
+        # Initialize an UrllibClient for making HTTP requests
+        client = UrllibClient()
 
-        cookie.load(vars(client.get("https://rentry.co"))["headers"]["Set-Cookie"])
-        csrftoken = cookie["csrftoken"].value
+        # Get CSRF token for authentication
+        csrf_token = client.get_csrf_token()
 
+        # Prepare payload for the POST request
         payload = {
-            "csrfmiddlewaretoken": csrftoken,
+            "csrfmiddlewaretoken": csrf_token,
             "text": text,
         }
 
-        return json_loads(
-            client.post("https://rentry.co/api/new", payload, headers=_headers).data
-        )
-
+        # Perform the POST request to create a new entry
+        return json_loads(client.post(API_NEW_ENDPOINT, payload, headers=_HEADERS).data)
 
 class RentryImport(QDialog):
```

Comment:
why ? right now its just a basic QDialog
from PySide6.QtWidgets import QDialog

Reply:
There are two principles of object-oriented programming that we need to take into consideration in this situation.

1. The single-responsibility principle.
2. Composition versus inheritance.

The single-responsibility principle, which is fundamental to object-oriented programming, says that classes should have just one job or area of responsibility. Here you have a class that is responsible for a GUI element (the dialog box) and that is also responsible for talking to the Rentry web site, downloading content and parsing it.

This class should be separated into two classes: One class that is responsible for downloading from the Rentry site and parsing the data, and a second class that is responsible for the GUI element.

As for composition versus inheritance, we usually talk in terms of _has a_ versus _is a._ Inheritance is an _is a_ relationship: Apple _is a_ Fruit; Employee _is a_ Person. Where an _is a_ relationship exists, it's correct to use inheritance.

Composition represents a _has a_ relationship. It would be a programming error for the Employee class to inherit from the EmailAddress class, because an employee _is not an_ email address. Instead an employee _has an_ email address, so the Employee class should contain a reference to an EmailAddress object as an attribute.

Here you're inheriting from QDialog, saying that this class _is a_ QDialog. But it isn't. It downloads content from Rentry; that's its job. This class can _have a_ QDialog (or more properly, a QDialog subclass) but it's wrong to say it _is a_ QDialog. That's a programming error.

Reply:
yeah i see your point but unless there are more places that we would be needing QDialog entirely as a class i dont want to make it so maybe you can do it after the merge and show me what you mean i am happy to learn

Reply:
Yeah, all right.